include/particle.tcc

```diff
@@ -312,6 +313,70 @@ bool mpm::Particle<Tdim, Tnphases>::compute_stress(unsigned phase) {
   return status;
 }
 
+// Compute pressure
+template <unsigned Tdim, unsigned Tnphases>
+bool mpm::Particle<Tdim, Tnphases>::compute_pressure(unsigned phase,
+                                                     double dt) {
+  bool status = true;
+  double pressure, dpressure;
+  double bulk_modulus, youngs_modulus, poisson_ratio, dvolumetric_strain;
+
+  // Strain rate for reduced integration
```

Comment:
Please make sure a function does one task, move `strain_rate_centroid_` to `compute_strain` function

Reply:
So we add another private variable then?

Reply:
And another public function called `compute_strain_centroid`?

Reply:
If we do so, we can also just output the `strain_rate_centroid_` and pressure calculation can be done in `material` and we can feed it back to `particle`... So `particle` would call `compute_pressure` and it will call on the pressure from `material`. This is more aligned with what we are doing for the stress. Would this be OK?